codes/quantum/gauge_stabilizer

```diff
@@ -0,0 +1,76 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: gauge_stabilizer
+physical: qubits
+logical: qubits
+
+name: 'Gauge Stabilizer Codes'
+
+introduced: '\cite{arXiv:quant-ph/0508131}'
+
+description: 'Choose \(2n\) operators \{ X_j',Z_j'\}_{j=1}^n' from \(\mathcal{P}_n\), the Pauli group on \(n\) qubits, such that they obey the same commutation relations as the regular \(n\)-qubit Pauli generators \( \{X_j,Z_j\}_{j=1}^n \) (the subscript on these latter operators indicates the single qubit the Pauli matrix acts on). The primed operators might act on more than one physical (or \textit{bare}) qubit but they behave as if they acted only on a single qubit. WLOG we can choose a stabilizer group as \( \mathcal{S} = \langle Z_1,\dots,Z_s \rangle \). It follows that the normalizer of \(\mathcal{S} \) is \( N(\mathcal{S}) = \langle i,Z_1',\dots,Z_n',X_{s+1}',\dots,X_n' \rangle \). We now choose a gauge group as \( \mathcal{G} = \langle i, Z_1',\dots,Z_s', X_{s+1}',Z_{s+1}', \dots, X_{s+r}',Z_{s+r}' \rangle \) with \( s + r \leq n \). The logical group is choosen as \( \mathcal{L} = N(\mathcal{S})/\mathcal{G} \simeq \langle X_{s+r+1}',Z_{s+r+1}', \dots, X_n',Z_n' \rangle \). Now the codespace \( C \) is as usual the \(+1\) eigenspace of the stabilizer \( \mathcal{S} \). But the gauge and logical groups have further decomposed this space into \( C = A \otimes B  \simeq (\mathbb{C}^2)^{\otimes k} \otimes (\mathbb{C}^2)^{\otimes r} \). Thus the Hilbert space is partitioned into 3 sets; \(k\) logical qubits, \(r\) gauge qubits, and \(s\) stabilizer qubits, with \(s+r+k=n\). '
```

Comment:
Nice technical defn. Could you prepend with an intuitive sentence? Here is an excerpt from Gottesman to help: "There are a couple of dierent ways of looking at subsystem coding, but they all move beyond this picture. In one view, we instead are encoding single states as subspaces. Another way of looking at it is that we have a family of QECCs and we sometimes use one code and sometimes another code in the family. A third way of looking at it is that we have a QECC with some extra encoded qubits, but we don't care about the values of the extra logical qubits. They are called
gauge qubits."

Reply:
Also, add in the very beginning of description: "Also called subsystem stabilizer code."